src/main/java/mate/academy/springbootintro/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,91 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/cart")
+@RequiredArgsConstructor
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @GetMapping
+    @PreAuthorize(value = "hasRole('USER')")
+    @Operation(
+            summary = "Get shopping cart",
+            description = "Getting shopping cart for current user"
+    )
+    public ShoppingCardDto getShoppingCard(Authentication authentication) {
+        return shoppingCartService.findShoppingCardByUserId(getCurrentUserId(authentication));
+    }
+
+    @PostMapping
+    @PreAuthorize(value = "hasRole('USER')")
+    @Operation(
+            summary = "Add item carts to shopping cart",
+            description = "Adding item carts to users shopping cart"
+    )
+    public ShoppingCardDto addItemToCart(
+            @RequestBody @Valid CreateCartItemRequestDto cartItemRequestDto,
+            Authentication authentication
+    ) {
+        return shoppingCartService.addCartItem(
+                cartItemRequestDto,
+                getCurrentUserId(authentication)
+        );
+    }
+
+    @PutMapping(value = "/items/{cartItemId}")
+    @PreAuthorize(value = "hasRole('USER')")
+    @Operation(
+            summary = "Update quantity",
+            description = "Updating products quantity in item carts"
+    )
+    public ShoppingCardDto updateQuantity(
+            @PathVariable @Positive Long cartItemId,
+            @RequestBody @Valid UpdateCartItemRequestDto updateCartItemRequestDto,
+            Authentication authentication
+    ) {
+        return shoppingCartService.updateQuantityInCartItem(
+                cartItemId,
+                updateCartItemRequestDto,
+                getCurrentUserId(authentication)
+        );
+    }
+
+    @DeleteMapping(value = "/items/{cartItemId}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @PreAuthorize(value = "hasRole('USER')")
+    @Operation(summary = "Delete item cart", description = "Deleting item carts in shopping cart")
+    public void deleteItemCart(Long cartItemId, Authentication authentication) {
+        shoppingCartService.deleteCartItem(cartItemId, getCurrentUserId(authentication));
+    }
+
+    private Long getCurrentUserId(Authentication authentication) {
+        authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
```suggestion
```

Reply:
Why you need this ?
`authentication = SecurityContextHolder.getContext().getAuthentication();`